rust-connectors/common/src/opt.rs

```diff
@@ -399,6 +375,36 @@ impl CommonConnectorOpt {
             }
         }
     }
+
+    #[cfg(any(feature = "sink", feature = "source"))]
+    async fn get_aggregate_initial_value(&self, fluvio: &Fluvio) -> anyhow::Result<Vec<u8>> {
+        use tokio_stream::StreamExt;
+        if let Some(initial_value) = &self.smartmodule_common.aggregate_initial_value {
+            if initial_value == "use-last" {
+                let consumer = fluvio
+                    .partition_consumer(self.fluvio_topic.clone(), 0)
+                    .await?;
+                let stream = consumer.stream(fluvio::Offset::from_end(1)).await?;
+                let timeout = stream.timeout(Duration::from_millis(3000));
```

Comment:
can we use env var to configure this?

Reply:
I think a `const` might be a bit more appropriate. I just don't think we'll need to change the timeout when attempting to get the last record in the stream. A common flow here would be that a connector using an aggregate (and an initial value of `use-last`), the user of this should want their connector to fail relatively quickly if there's nothing in the topic.

Reply:
I just moved this code from one place to another to avoid compilation warnings about unused code when not all features are enabled. So, it is not related to the changes in the PR. If we want to change something here, we should do it in another PR.

Reply:
fair enough.  can you create issue for that? I will close this comment